kube-client/src/client/auth/mod.rs

```diff
@@ -89,14 +94,34 @@ pub(crate) enum Auth {
 // - gcp: command based token source (exec)
 // - gcp: application credential based token source (requires `oauth` feature)
 #[derive(Debug, Clone)]
-pub(crate) enum RefreshableToken {
+#[doc(hidden)]
+pub enum RefreshableToken {
```

Comment:
Had to make this `pub` because it's part of `AuthLayer::Service`.

Reply:
I think it's a reasonable tradeoff here between not wanting to make things public, so we hide it instead. So have no real problems with this pr, but I don't really see how `AuthLayer::Service` forces this. It looks like `AuthLayer` contains stuff that is private, and it doesn't feel like the generic associated type should be exposed because of this. But haven't played around with this.

Reply:
It's because of `impl<S> Layer<S> for AuthLayer`. The `Layer`'s associated type `Service` is public.

```
error[E0446]: crate-private type `RefreshableToken` in public interface
   --> kube-client/src/client/middleware/mod.rs:14:5
    |
14  | /     type Service = Either<
15  | |         <AddAuthorizationLayer as Layer<S>>::Service,
16  | |         <AsyncFilterLayer<RefreshableToken> as Layer<S>>::Service,
17  | |     >;
    | |______^ can't leak crate-private type
    |
   ::: kube-client/src/client/auth/mod.rs:101:1
    |
101 |   pub(crate) enum RefreshableToken {
    |   -------------------------------- `RefreshableToken` declared as crate-private
```

Reply:
I'm confused now. `RefreshTokenLayer` was crate-private too, and it was fine.

Reply:
Hah, yeah, that's what I saw.